### PR TITLE
Add allowlisted_modules and deprecate whitelisted_modules in license-lint config

### PR DESCRIPTION
**Pull request by tbarrella**
As part of https://github.com/istio/istio/issues/25381

### CODE REVIEW

File: cmd/license-lint/config.go [hunk 1]
```diff
@@ -32,6 +32,8 @@ type rawConfig struct {
 	RestrictedLicenses []string `json:"restricted_licenses"`
 
 	// modules that get completely ignored during analysis
+	AllowlistedModules []string `json:"allowlisted_modules"`
+	// Deprecated. TODO(tbarrella): Clean up
 	WhitelistedModules []string `json:"whitelisted_modules"`
```
Comment (carolynhu) [inline cmd/license-lint/config.go:37]: Does` AllowlistedModules` have the same meaning of `WhitelistedModules`?

Comment (tbarrella) [reply cmd/license-lint/config.go:37]: Yeah, it's meant to replace it. It seemed like I needed to stage the API change to not temporarily break the linting across repos

Comment (howardjohn) [reply cmd/license-lint/config.go:37]: Some more info on the breakage: there are two parts that are unfortunately separated: the docker image that the CI runs, and the actual linter config. There is no way to change the docker iamge CI uses as part of a PR, its all defined in a separate way in istio/test-infra. So if you do breaking changes there is some downtime/synchronization required

**Comment by tbarrella** [comment]
To test I removed the `whitelisted_modules` key from common/config/license-lint.yml and `make lint-licenses` ran without panicking

**Pull request merged**